support/reftest/reftest

```diff
@@ -0,0 +1,249 @@
+        config = yaml.load(f, yaml.SafeLoader)
+    return RefTestConfig(config["prod-cdn-url"], config["test_data"],)
+
+
+def download_to_local(url, key_path, cert_path, cacert_path):
+    with requests.get(
+        url,
+        cert=(cert_path, key_path),
+        verify=cacert_path,
+        stream=True,
+        timeout=(30, 30),
+    ) as r:
+        r.raise_for_status()
+        total_size = int(r.headers.get("content-length", 0))
+        t = tqdm(desc=url, total=total_size, unit="iB", unit_scale=True)
+
+        # the file is deleted as soon as it is closed.
+        temp_file = tempfile.NamedTemporaryFile(delete=True)
+        for chunk in r.iter_content(chunk_size=8192):
+            if chunk:  # filter out keep-alive new chunks
+                t.update(len(chunk))
+                temp_file.write(chunk)
+                temp_file.flush()
+        t.close()
+        return temp_file
+
+
+# verify cdn data checksum mathch test data checksum
+def verify_sha256_checksum(temp_file_path, test_data_checksum):
+    if os.path.isfile(temp_file_path):
+        sha256 = hashlib.sha256()
+        with open(temp_file_path, "rb") as binary:
+            while True:
+                chunk = binary.read(4096)
+                if not chunk:
+                    break
+                sha256.update(chunk)
+
+        cdn_data_checksum = sha256.hexdigest()
+
+        if cdn_data_checksum == test_data_checksum:
+            return True
+
+    return False
+
+
+def prepare(db, s3, config, opt):
+    for item in config.test_data:
+        url = config.prod_cdn_url + item["path"]
+        # download test data to local with a name "NamedTemporaryFile"
+        temp_file = download_to_local(url, opt.key, opt.cert, opt.cacert)
+
+        # checksum verify
+        if not verify_sha256_checksum(temp_file.name, item["sha256"]):
+            print("{} verify checksum failed".format(item["path"]))
+            return False
+
+        # push test data to s3 and dynamodb
+        s3.upload_from_localfile(
+            temp_file.name, item["sha256"], item["content-type"]
+        )
+        db.put_item(item["path"], item["sha256"], opt.release_date)
+
+        # delete the NamedTemporaryFile
+        temp_file.close()
+
+    return True
+
+
+def main():
+    config = load_config()
+    opt = parse_args()
+
+    session = boto3.Session(
+        aws_access_key_id=opt.aws_access_id,
+        aws_secret_access_key=opt.aws_access_key,
+        aws_session_token=opt.aws_session_token,
+        region_name=opt.default_region,
+    )
+
+    db_client = DBHandler(table=opt.table, session=session)
+    s3_client = S3Handler(bucket=opt.bucket, session=session)
+
+    res = False
+    if opt.command == "prepare":
+        res = prepare(db_client, s3_client, config, opt)
+
+    if res:
+        print("{} operation has finished successfully!".format(opt.command))
+    else:
+        print("Fatal error: {} operation is terminated".format(opt.command))
```

Comment:
Can you please make sure it exits with non-zero exit code in this case?

Otherwise if someone tries to do something like `support/reftest/reftest && run_tests`, it'll go ahead and start run_tests even if the reftest command failed...